crates/brioche/src/check.rs

```diff
@@ -1,51 +1,109 @@
+use std::path::PathBuf;
 use std::process::ExitCode;
+use std::vec;
 
+use brioche_core::project::ProjectHash;
+use brioche_core::project::Projects;
 use brioche_core::reporter::ConsoleReporterKind;
+use brioche_core::Brioche;
 use clap::Parser;
 use tracing::Instrument;
 
+use crate::consolidate_result;
+
 #[derive(Debug, Parser)]
 pub struct CheckArgs {
     #[command(flatten)]
-    project: super::ProjectArgs,
+    project: super::MultipleProjectArgs,
 }
 
 pub async fn check(args: CheckArgs) -> anyhow::Result<ExitCode> {
     let (reporter, mut guard) =
         brioche_core::reporter::start_console_reporter(ConsoleReporterKind::Auto)?;
 
-    let brioche = brioche_core::BriocheBuilder::new(reporter).build().await?;
+    let brioche = brioche_core::BriocheBuilder::new(reporter.clone())
+        .build()
+        .await?;
     let projects = brioche_core::project::Projects::default();
 
-    let check_future = async {
-        let project_hash = super::load_project(&brioche, &projects, &args.project).await?;
+    let mut error_result = Option::None;
 
-        let num_lockfiles_updated = projects.commit_dirty_lockfiles().await?;
-        if num_lockfiles_updated > 0 {
-            tracing::info!(num_lockfiles_updated, "updated lockfiles");
-        }
+    // Handle the case where no projects and no registries are specified
+    let projects_path = if args.project.project.is_empty() && args.project.registry.is_empty() {
+        vec![PathBuf::from(".")]
+    } else {
+        args.project.project
+    };
 
-        let checked = brioche_core::script::check::check(&brioche, &projects, project_hash).await?;
+    // Loop over the projects
+    for project_path in projects_path {
+        let project_name = format!("project '{name}'", name = project_path.display());
 
-        guard.shutdown_console().await;
+        match projects.load(&brioche, &project_path, true).await {
+            Ok(project_hash) => {
+                let result = run_check(&brioche, &projects, project_hash, &project_name).await;
+                consolidate_result(&project_name, result, &mut error_result);
+            }
+            Err(e) => {
+                consolidate_result(&project_name, Err(e), &mut error_result);
+            }
+        }
+    }
 
-        let result = checked.ensure_ok(brioche_core::script::check::DiagnosticLevel::Message);
+    // Loop over the registries
```

Comment:
Nitpick: rewording. `-r` takes a registry project (that is, a "project that should get downloaded from the registry") not a registry itself (and the flag is named `registry` to keep it shorter... although renaming the struct field while keeping the flag the same might be a good idea)

```suggestion
    // Loop over the registry projects
```

Reply:
I updated the registry argument to registry_project, and I added a clap identifier to not change the CLI interface.